base/deprecated.jl

```diff
@@ -1138,6 +1138,10 @@ import .Random: srand
 @deprecate srand(filename::AbstractString, n::Integer=4) srand(read!(filename, Vector{UInt32}(uninitialized, Int(n))))
 @deprecate MersenneTwister(filename::AbstractString)  srand(MersenneTwister(0), read!(filename, Vector{UInt32}(uninitialized, Int(4))))
 
+# PR #25429
+@deprecate rand(r::AbstractRNG, dims::Dims) rand(r, Float64, dims)
```

Comment:
Does this one really need to be deprecated? I thought a dims tuple as the second argument was ok (and there are still some other methods with that).

Reply:
in `rand([rng], [S], [dims])`, the conflict is when using `t::Tuple` either for `S` or for `dims`, whether `rng` is specified or not. In `rand(rng, (1, 2, 3))`, the conflict is the same as in `rand((1, 2, 3))`. A tuple as the second argument is ok when `rng` is not specifed, as in `rand(Float64, (1, 2, 3))`.

> and there are still some other methods with that

I must have missed that, but no test catched this apparently...

Reply:
I see. In that case this seems fine as-is.